### PR TITLE
migrate to click

### PR DESCRIPTION
**Pull request by dineshyv**
Tests:
```
llama-stack-client models list
+----------------------+--------------------+---------------------------------------------+------------+
| identifier           | provider_id        | provider_resource_id                        | metadata   |
+======================+====================+=============================================+============+
| Llama3.1-8B-Instruct | remote::together-0 | meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo | {}         |
+----------------------+--------------------+---------------------------------------------+------------+
❯ llama-stack-client providers list
+-----------+--------------------------+------------------------+
| API       | Provider ID              | Provider Type          |
+===========+==========================+========================+
| inference | remote::together-0       | remote::together       |
+-----------+--------------------------+------------------------+
| memory    | inline::faiss-0          | inline::faiss          |
+-----------+--------------------------+------------------------+
| memory    | remote::weaviate-1       | remote::weaviate       |
+-----------+--------------------------+------------------------+
| safety    | inline::llama-guard-0    | inline::llama-guard    |
+-----------+--------------------------+------------------------+
| agents    | inline::meta-reference-0 | inline::meta-reference |
+-----------+--------------------------+------------------------+
| telemetry | inline::meta-reference-0 | inline::meta-reference |
+-----------+--------------------------+------------------------+
❯ llama-stack-client models get Llama3.1-8B-Instruct
+----------------------+------------+--------------------+---------------------------------------------+--------+
| identifier           | metadata   | provider_id        | provider_resource_id                        | type   |
+======================+============+====================+=============================================+========+
| Llama3.1-8B-Instruct | {}         | remote::together-0 | meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo | model  |
+----------------------+------------+--------------------+---------------------------------------------+--------+
```

### CODE REVIEW

**Comment by yanxi0830** [comment]
https://github.com/meta-llama/llama-stack-client-python/blob/main/docs/cli_reference.md

Could you attach the test runs on all these commands?

File: src/llama_stack_client/lib/cli/models/list.py [hunk 1]
```diff
@@ -4,43 +4,17 @@
...
-
-        client = LlamaStackClient(
-            base_url=args.endpoint,
-        )
-
-        headers = ["identifier", "llama_model", "provider_id", "metadata"]
-        response = client.models.list()
-        if response:
-            print_table_from_response(response, headers)
+    headers = ["identifier", "provider_id", "provider_resource_id", "metadata"]
```
Comment (yanxi0830) [inline src/llama_stack_client/lib/cli/models/list.py:17]: Thanks! Was also fixing a similar issue.

File: src/llama_stack_client/lib/cli/configure.py [hunk 2]
```diff
@@ -1,94 +0,0 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (yanxi0830) [inline src/llama_stack_client/lib/cli/configure.py:1]: Could we still keep configure? Or if you have a proposal for the alternative for using the CLI interface?
```
llama stack configure --endpoint 
```

Comment (dineshyv) [reply src/llama_stack_client/lib/cli/configure.py]: done.

**Pull request merged**